pkg/actuators/machine/actuator.go

```diff
@@ -105,140 +103,121 @@ func (a *Actuator) handleMachineError(machine *machinev1.Machine, err error, eve
+	err = a.setStatus(machine, instance)
+	if err != nil {
+		return fmt.Errorf("%s: failed to set machine status: %v", machine.Name, err)
+	}
+
+	err = a.patchMachine(context, machine, machineToBePatched)
+	if err != nil {
+		return fmt.Errorf("%s: failed to patch machine: %v", machine.Name, err)
+	}
+
+	return a.requeueIfInstancePending(machine.Name, instance)
 }
 
-func (a *Actuator) setMachineCloudProviderSpecifics(machine *machinev1.Machine, instance *ec2.Instance) (*machinev1.Machine, error) {
+func (a *Actuator) setMachineCloudProviderSpecifics(machine *machinev1.Machine, instance *ec2.Instance) error {
 	if instance == nil {
-		return machine, nil
+		return nil
 	}
-	machineCopy := machine.DeepCopy()
 
-	if machineCopy.Labels == nil {
-		machineCopy.Labels = make(map[string]string)
+	if machine.Labels == nil {
+		machine.Labels = make(map[string]string)
 	}
 
-	if machineCopy.Annotations == nil {
-		machineCopy.Annotations = make(map[string]string)
+	if machine.Annotations == nil {
+		machine.Annotations = make(map[string]string)
 	}
 
 	// Reaching to machine provider config since the region is not directly
 	// providing by *ec2.Instance object
 	machineProviderConfig, err := providerConfigFromMachine(machine, a.codec)
 	if err != nil {
-		return nil, fmt.Errorf("error decoding MachineProviderConfig: %v", err)
+		return fmt.Errorf("error decoding MachineProviderConfig: %v", err)
 	}
 
-	machineCopy.Labels[machinecontroller.MachineRegionLabelName] = machineProviderConfig.Placement.Region
+	machine.Labels[machinecontroller.MachineRegionLabelName] = machineProviderConfig.Placement.Region
 
 	if instance.Placement != nil {
-		machineCopy.Labels[machinecontroller.MachineAZLabelName] = aws.StringValue(instance.Placement.AvailabilityZone)
+		machine.Labels[machinecontroller.MachineAZLabelName] = aws.StringValue(instance.Placement.AvailabilityZone)
 	}
 
 	if instance.InstanceType != nil {
-		machineCopy.Labels[machinecontroller.MachineInstanceTypeLabelName] = aws.StringValue(instance.InstanceType)
+		machine.Labels[machinecontroller.MachineInstanceTypeLabelName] = aws.StringValue(instance.InstanceType)
 	}
 
 	if instance.State != nil && instance.State.Name != nil {
-		machineCopy.Annotations[machinecontroller.MachineInstanceStateAnnotationName] = aws.StringValue(instance.State.Name)
+		machine.Annotations[machinecontroller.MachineInstanceStateAnnotationName] = aws.StringValue(instance.State.Name)
 	}
 
-	if err := a.client.Update(context.Background(), machineCopy); err != nil {
-		return nil, fmt.Errorf("%s: error updating machine spec: %v", machine.Name, err)
-	}
-
-	return machineCopy, nil
+	return nil
 }
 
-// updateProviderID adds providerID in the machine spec
-func (a *Actuator) updateProviderID(machine *machinev1.Machine, instance *ec2.Instance) (*machinev1.Machine, error) {
+// setProviderID adds providerID in the machine spec
+func (a *Actuator) setProviderID(machine *machinev1.Machine, instance *ec2.Instance) error {
```

Comment:
This method doesn't seem to ever actually return an error. Seems like we could just drop that return type. Though, if we want to keep it as-is just to have a common API among setter methods, that's fine.